### ISSUE TITLE
Support room IDs in new search experience

### ISSUE THREAD

**Issue by HarHarLinks** [vector-im/element-web#22500]
#### What would you like to do?
Enter a room ID (not alias!) such as `!0123456789abcdef:example.org` into the search bar and navigate to the respective room.

This should automatically work with room IDs encoded in matrix.to URLs the same.

### Additional context

- #15840: asks for UI to do a similar thing
- #21695: asks for support of matrix.to URLs that use aliases in the new search experience

**Comment by aaronraimist** [comment]
> OOI, does the same problem apply when a via server is provided when using a URL?

Yes the same problem exists there and the error message has been awful for a long time. https://github.com/matrix-org/matrix-react-sdk/pull/6048

**Comment by t3chguy** [comment]
@HarHarLinks the server is only there for uniqueness and is not guaranteed to even exist

**Comment by HarHarLinks** [comment]
> not guaranteed to even exist

So is any given via server, hence I don't understand why that would be a strong counterargument.

**Comment by t3chguy** [comment]
Clients are meant to treat room IDs as entirely opaque, so if anyone is to try the server in the room ID as a via server then it should be your HS.

**Comment by HarHarLinks** [comment]
Fair enough, though how would the via server given in a URL be handled differently?
